src/openrct2/actions/WallPlaceAction.cpp

```diff
@@ -427,12 +419,11 @@ GameActions::Result::Ptr WallPlaceAction::Execute() const
         wallElement->SetTertiaryColour(_tertiaryColour);
     }
 
-    if (GetFlags() & GAME_COMMAND_FLAG_GHOST)
-    {
-        wallElement->SetGhost(true);
-    }
+    wallElement->SetGhost(GetFlags() & GAME_COMMAND_FLAG_GHOST);
 
-    res->tileElement = tileElement;
+    res->tileElement = wallElement->as<TileElement>();
+
+    map_animation_create(MAP_ANIMATION_TYPE_WALL, targetLoc);
```

Comment:
Any reason you moved this line?

Reply:
It looked a bit out of place between the creation and initialization of the tile element.